lib/IMAP/PreviewEnhancer.php

```diff
@@ -68,7 +68,7 @@ public function __construct(IMAPClientFactory $clientFactory,
 	 */
 	public function process(Account $account, Mailbox $mailbox, array $messages): array {
 		$needAnalyze = array_reduce($messages, static function (array $carry, Message $message) {
-			if ($message->getStructureAnalyzed()) {
+			if ($message->getStructureAnalyzed() && $message->getDkimStatus() !== Message::DKIM_STATUS_PENDING) {
```

Comment:
Better: A migration to reset structure analyzed. But validating the dkim signatur may take a while.